src/main/java/it/mulders/mcs/cli/Cli.java

```diff
@@ -31,68 +35,66 @@ public class Cli {
+            "Otherwise, the search term is considered a wildcard search"
+        }
     )
-    public class SearchCommand implements Callable<Integer> {
-        @CommandLine.Parameters(
-                arity = "1..n",
-                description = {
-                        "What to search for.",
-                        "If the search term contains a colon ( : ), it is considered a literal groupId and artifactId",
-                        "Otherwise, the search term is considered a wildcard search"
-                }
-        )
-        private String[] query;
+    private String[] query;
 
-        @CommandLine.Option(
-                names = { "-l", "--limit" },
-                description = "Show <count> results",
-                paramLabel = "<count>"
-        )
-        private Integer limit;
+    @CommandLine.Option(
+        names = { "-l", "--limit" },
+        description = "Show <count> results",
+        paramLabel = "<count>"
+    )
+    private Integer limit;
 
-        @CommandLine.Option(
-                names = { "-f", "--format" },
-                description = """
+    @CommandLine.Option(
+        names = { "-f", "--format" },
+        description = """
                         Show result in <type> format
                         Supported types are:
                           maven, gradle, gradle-short, gradle-kotlin, sbt, ivy, grape, leiningen, buildr
                         """,
-                paramLabel = "<type>"
-        )
-        private String responseFormat;
+        paramLabel = "<type>"
+    )
+    private String responseFormat;
 
-        @CommandLine.Option(
-                names = { "-s", "--show-vulnerabilities" },
-                description = "Show reported security vulnerabilities",
-                paramLabel = "<vulnerabilities>"
-        )
-        private boolean showVulnerabilities;
+    @CommandLine.Option(
+        names = { "-s", "--show-vulnerabilities" },
+        description = "Show reported security vulnerabilities",
+        paramLabel = "<vulnerabilities>"
+    )
+    private boolean showVulnerabilities;
 
-        @Override
-        public Integer call() {
-            var combinedQuery = String.join(" ", query);
-            System.out.printf("Searching for %s...%n", combinedQuery);
-            var searchQuery = SearchQuery.search(combinedQuery)
-                    .withLimit(this.limit)
-                    .build();
 
-            CoordinatePrinter coordinatePrinter = FormatType.providePrinter(responseFormat);
-            var searchCommandHandler = new SearchCommandHandler(coordinatePrinter, showVulnerabilities);
-            searchCommandHandler.search(searchQuery);
-            return 0;
+    @Spec
+    CommandSpec spec;
+
+    @Override
+    public Integer call() {
+        if (Objects.isNull(query)) {
+            spec.commandLine().usage(System.err);
+            System.exit(1);
         }
```

Comment:
Added because of this [change](https://github.com/mthmulders/mcs/pull/289/files#r1425313469). This explicit validation means the commands `mcs` and previously `mcs search` (missing the query) both do the same thing e.g. print usage and exit.